internal/access_provider_resource.go

```diff
@@ -59,12 +61,14 @@ type AccessProviderModel[T any] interface {
 
 type ReadHook[T any, ApModel AccessProviderModel[T]] func(ctx context.Context, client *sdk.RaitoClient, data ApModel) diag.Diagnostics
 type ValidationHook[T any, ApModel AccessProviderModel[T]] func(ctx context.Context, data ApModel) diag.Diagnostics
+type PlanModifierHook[T any, ApModel AccessProviderModel[T]] func(ctx context.Context, data ApModel) (ApModel, diag.Diagnostics)
 
 type AccessProviderResource[T any, ApModel AccessProviderModel[T]] struct {
 	client *sdk.RaitoClient
 
 	readHooks      []ReadHook[T, ApModel]
 	validationHoos []ValidationHook[T, ApModel]
+	planModifier   []PlanModifierHook[T, ApModel]
```

Comment:
Maybe fix the name of validationHoos to ... Hooks :) This is not woke ;)

Reply:
Will do this in a followup PR